x/rollapp/types/rollapp.go

```diff
@@ -0,0 +1,63 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (r Rollapp) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(r.Creator)
+	if err != nil {
+		return errorsmod.Wrap(err, ErrInvalidCreatorAddress.Error())
+	}
+	if r.GetMaxSequencers() == 0 {
+		return ErrInvalidMaxSequencers
+	}
+
+	// validate rollappId
+	_, err = ParseChainID(r.RollappId)
+	if err != nil {
+		return err
+	}
+
+	// verifies that there's no duplicate address in PermissionedAddresses
+	// and addresses are in Bech32 format
+	permissionedAddresses := r.GetPermissionedAddresses()
+	if len(permissionedAddresses) > 0 {
+		duplicateAddresses := make(map[string]bool)
+		for _, item := range permissionedAddresses {
+			// check if the item/element exist in the duplicateAddresses map
+			_, exist := duplicateAddresses[item]
+			if exist {
+				return errorsmod.Wrapf(ErrPermissionedAddressesDuplicate, "address: %s", item)
+			}
+			// check Bech32 format
+			if _, err := sdk.AccAddressFromBech32(item); err != nil {
+				return errorsmod.Wrapf(ErrInvalidPermissionedAddress, "invalid permissioned address: %s", err)
+			}
+			// mark as exist
+			duplicateAddresses[item] = true
+		}
+	}
```

Comment:
this could all be shortened by a generic dedupe function, and https://pkg.go.dev/slices#ContainsFunc or https://pkg.go.dev/slices#IndexFunc

psuedo:

```
if len(permissioned) != len(deduped(permissioned)) {
..}
if i := slices.indexFunc(permissioned, func(x){
_, err := sdk.AccAddressFromBech32(x)
return err!=nil
}
}) {..}

```

something like that

Reply:
(just a suggestion, it's fine)

Reply:
opened separate issue for it 
https://github.com/dymensionxyz/dymension/issues/727